oauthproxy.go

```diff
@@ -773,6 +777,11 @@ func (p *OAuthProxy) AuthenticateOnly(rw http.ResponseWriter, req *http.Request)
 		return
 	}
 
+	if len(p.AllowedGroups) > 0 && !p.validateGroups(session.Groups) {
```

Comment:
Can we rollup all of this logic together in a `p.authorize(session)` function?

Reply:
@JoelSpeed Looking for a second opinion on this one: Should we do group authorization in AuthenticateOnly in addition to Proxy?

Why I'm torn:

In today's world, where group authZ is global - this makes sense to add it in here as well.

In V7 structured configs where we can have group claim based AuthZ rules on a per path basis, paths aren't available to users accessing this function via nginx `auth_request` -- those users would need to use the x-auth-request header back at the nginx layer to get fine-grained authZ.

So should we leave this function like its name says, pure AuthN -- no AuthZ?

Reply:
`getAuthenticatedSession` has the email validator check in it (thank goodness, otherwise using something like the google provider + nginx auth_request would have a big hole 😅  )

`getAuthenticatedSession`  is used in `UserInfo`, `AuthenticateOnly` & `Proxy` -- `UserInfo` was the key one that gave me pause from just including the AuthZ check completely in `getAuthenticatedSession`.

I'm thinking in a V6 global group authZ context -- it makes sense to just put this logic into `getAuthenticatedSession`.  We can work out the refactor in a V7 world once per path AuthZ rules are a thing later?